### DISCUSSION TITLE
maybe a neon aegis128l implement?

### DISCUSSION THREAD

**Discussion by calvin2021y** [jedisct1/libsodium#1265] [Ideas]
the soft aegis128l is too slow, maybe add neon asm implement to speed it up ?

If so, we can use aegis128l replace cacha20 for all mobile use case.  

not need as fast as chache20, just like 1/2 or 1/3 will be good enough.

**Comment by jedisct1** [comment]
There's already a Neon implementation. It was added at the same time as the Intel one.

And even on ARM, it massively faster than ChaCha20 and AES-GCM.

**Comment by jedisct1** [comment]
Oh, my bad, you probably mean just NEON, for ARM CPUs that don't have AES instructions at all.

AES doesn't benefit from NEON instructions. If a CPU doesn't have AES instructions, Salsa and ChaCha are always faster.

**Comment by jedisct1** [comment]
If side channels are not an issue, you could also define `SOFTAES_STRIDE` to `256` (compile with `env CPPFLAGS=-DSOFTAES_STRIDE=256 ./configure ...`) in order to speed up soft AES.
